### PR TITLE
Added integration tests for sorting

### PR DESCRIPTION
**Pull request by PascalSenn**
Depends on #2311

### CODE REVIEW

File: src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorBooleanTests.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+using HotChocolate.Tests;
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+
+    public class QueryableSortVisitorBooleanTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities = new[]{
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorBooleanTests.cs:12]: Formatting

private static readonly FooNullable[] _fooNullableEntities = new[]
{
    new FooNullable { Bar = true },
    new FooNullable { Bar = null },
    new FooNullable { Bar = false }
};

File: src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorComparableTests.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+using HotChocolate.Execution;
+using HotChocolate.Tests;
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+    public class QueryableSortVisitorComparableTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities = new[]{
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorComparableTests.cs:11]: Formatting

File: src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorEnumTests.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+using Xunit;
+
+namespace HotChocolate.Data.Sorting.Expressions
+{
+    public class QueryableSortVisitorEnumTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities =
+        {
+            new Foo {BarEnum = FooEnum.BAR},
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorEnumTests.cs:13]: Formatting:

new Foo { BarEnum = FooEnum.BAR },

File: src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorStringTests.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+using HotChocolate.Tests;
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+    public class QueryableSortVisitorStringTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities =
+            new[] {new Foo {Bar = "testatest"}, new Foo {Bar = "testbtest"}};
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.InMemory.Tests/QueryableSortVisitorStringTests.cs:12]: Formatting

File: src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorBooleanTests.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+
+    public class QueryableSortVisitorBooleanTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities = new[]{
+            new Foo { Bar = true },
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorBooleanTests.cs:13]: Formatting

File: src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorComparableTests.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+using HotChocolate.Execution;
+using HotChocolate.Tests;
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+    public class QueryableSortVisitorComparableTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities = new[]{
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorComparableTests.cs:11]: Formatting

File: src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorEnumTests.cs [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+using Xunit;
+
+namespace HotChocolate.Data.Sorting.Expressions
+{
+    public class QueryableSortVisitorEnumTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities =
+        {
+            new Foo {BarEnum = FooEnum.BAR},
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorEnumTests.cs:13]: formatting 
new Foo {BarEnum = FooEnum.BAR},

File: src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorStringTests.cs [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+using HotChocolate.Tests;
+using Xunit;
+
+namespace HotChocolate.Data.Sorting
+{
+    public class QueryableSortVisitorStringTests
+        : IClassFixture<SchemaCache>
+    {
+        private static readonly Foo[] _fooEntities =
+            new[] {new Foo {Bar = "testatest"}, new Foo {Bar = "testbtest"}};
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Sorting.SqlLite.Tests/QueryableSortVisitorStringTests.cs:12]: formatting

**Review by michaelstaib — CHANGES_REQUESTED** [review]
There are some formatting issues left on this one.

**Comment by PascalSenn** [comment]
@michaelstaib  this one is ready

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**